pipenv/project.py

```diff
@@ -941,7 +967,10 @@ def remove_package_from_pipfile(self, package_name, dev=False):
         name = self.get_package_name_in_pipfile(package_name, dev)
         key = "dev-packages" if dev else "packages"
         p = self.parsed_pipfile
-        lines = [l for l in p[key].serialized().splitlines()]
+        try:
+            lines = [l for l in p.item(key).as_string().splitlines()]
+        except AttributeError:
+            lines = [l for l in p[key].serialized().splitlines()]
```

Comment:
What will happen if toml file contains line comments? I can't find the original bug 83af37a attempts to solve.
As discussed above, TOML do allow line comments.

Reply:
> # The normal toml parser can't handle deleting packages with preceding newlines

When I wrote this i actually meant it can't handle deleting packages from pipfiles where the preceding line is only made of a comment

I updated the comment on the PR in question to include the rest of the commits which weren't referenced explicitly, but the issue I was looking at was #2885

Reply:
In any event, this is all the reason why we use 4 different toml parsers (some handle inline comments correctly, some don't; some can't delete lines with preceding comments, some can)

Now I am seeing that contoml/prettytoml can't even handle writing files correctly and I'm pretty much done using them, and tomlkit is the only other one in python that I'm aware of that will retain the original document structure including outline/inline tables and comments. We are already using it in https://github.com/sarugaku/passa, https://github.com/sarugaku/requirementslib, https://github.com/sarugaku/plette, and pretty much everywhere else we have reimplemented pipfile parsing.  So far this is the only remaining issue I have with it, so I decided to go ahead and fix everything else.

Reply:
Ah, I didn't notice that bug in `contoml/prettytoml`. `tomlkit` seems to work quite well on this, making these lines a bit protective.

Reply:
Yeah I deleted this all locally but currently I’m not preserving the comment anymore